components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py

```diff
@@ -127,9 +129,17 @@ def _create_job_request(
 
         request["TransformResources"]["InstanceType"] = inputs.instance_type
         request["TransformResources"]["InstanceCount"] = inputs.instance_count
-        request["TransformResources"]["VolumeKmsKeyId"] = inputs.resource_encryption_key
-        request["DataProcessing"]["InputFilter"] = inputs.input_filter
-        request["DataProcessing"]["OutputFilter"] = inputs.output_filter
+
+        if inputs.resource_encryption_key:
```

Comment:
Can you add an else statement and delete `request["TransformResources"]["VolumeKmsKeyId"]` in that. 
Otherwise `request["TransformResources"]["VolumeKmsKeyId"]` dosent seem to get deleted because its part of the [request template](https://github.com/kubeflow/pipelines/blob/90841cbd65116319d8a8a17921b3bcd10b3cd88e/components/aws/sagemaker/batch_transform/src/sagemaker_transform_component.py#L84).

Reply:
@ananth102 
Thanks for your review.
I've fixed and rebased this PR. Please help to check again.

Reply:
Looks good! Cherry pick this to #8336

Reply:
@ananth102 
Done!